### ISSUE TITLE
[awGoToStep] seems to not be working (angular 9, archwizard 6)

### ISSUE THREAD

**Issue by theCrius** [madoar/angular-archwizard#299]
Hi there, first of all, thank you so much for this module. Really love that it's highly configurable, plenty of hooks and easy to style :)

I'm running in an issue that probably is just me not understanding how to properly use `awGoToStep`.

In short, what happens is that it doesn't work as I read it should in the readme.

I've put together a quick sandbox to show the issue, if you could point me to what am I doing wrong, i'd greatly appreciate it :)

The sandbox use the `stepIndex` but I tried also with the `stepId` and it didn't work as well.

https://stackblitz.com/edit/angular-ivy-9cviay

**Comment by madoar** [comment]
@theCrius it is nice to hear that you like `angular-archwizard` :)

About your issue:
`angular-archwizard` checks the to be performed navigation before executing it. This check entails that the wizard allows a navigation only if all previous steps of the destination step are completed. In simpler terms this means that a user can only navigate to the first uncompleted step and is unable to skip uncompleted steps.

If you want to allow a user to skip a specific step you can declare that step as `optional` by using the `awOptionalStep` directive on the wizard step element. This in turn allows the user to skip this specific step. 

Therefore the following changes should make your `awGoToStep` directive work:

-  update `typescript` to a newer version like ´3.8.0´ (this is required because otherwise the next change will lead to an error)
-  change your second step to:
   ```html
   <!-- STEP 2 -->
   <aw-wizard-step awOptionalStep>
     <div>Step 3</div>
     <button awPreviousStep>back</button>
     <button awNextStep>next</button>
   </aw-wizard-step>   
   ```

If you don't want that the optional step is shown with a green color by default you can change this by providing a custom stylesheet. For an example how this can be done see [here](https://madoar.github.io/angular-archwizard-demo/#/navigation-mode/free) and [here](https://github.com/madoar/angular-archwizard-demo/tree/develop/src/app/free-navigation-mode).

**Comment by theCrius** [comment]
That's making perfect sense @madoar. Thank you for the explanation.

I'm hiding the progress bar at the top and managing the progress through the steps with a series of if that shows the buttons to continue so for me it's good to just set every step as optional even if they are not really optional to the users :)

It may be worth to add this information (awOptionalStep) somewhere in the documentation?

Thanks again!

**Comment by madoar** [comment]
> It may be worth to add this information (awOptionalStep) somewhere in the documentation?

`[awOptionalStep]` is already documented in the README file, see [here](https://github.com/madoar/angular-archwizard#awoptionalstep). Do you think this should be extended?

**Comment by theCrius** [comment]
Maybe add a reference here? https://github.com/madoar/angular-archwizard#awgotostep

I read that section as I was interested in how the `awGoToStep` works and I didn't see mention of other steps before the one I want to skip to being required to be optional.

I also recognise that my use case may be unusual as I'm basically adding a button to jump the the last step as it shows the results of a questionnaire, if already completed, as a way to access them quickly by reusing the same UI.

**Comment by theCrius** [comment]
Created a pull request to save you from going and edit. I'm still new to this repo so I might have missed or misinterpreted how the constraint works, feel free to reject / edit the pull request.

https://github.com/madoar/angular-archwizard/pull/301
